client/token_claim_create.go

```diff
@@ -0,0 +1,147 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/kube"
+)
+
+func getSiteId(service *corev1.Service) string {
+	for _, ref := range service.ObjectMeta.OwnerReferences {
+		if ref.Name == "skupper-site" {
+			return string(ref.UID)
+		}
+	}
+	return ""
+}
+
+func getClaimsPort(service *corev1.Service) int32 {
+	for _, port := range service.Spec.Ports {
+		if port.Name == "claims" {
+			return port.Port
+		}
+	}
+	return 0
+}
+
+func (cli *VanClient) TokenClaimCreate(ctx context.Context, name string, password []byte, expiry time.Duration, uses int, secretFile string) error {
```

Comment:
I kept that in to match the other client methods (where it is also often unused at present). I think the thought was that it would be useful at some point. We can certainly revisit that, but I'd suggest doing so as a separate step and looking at the api as a whole.

Reply:
Thanks for the info.